contracts/BatchAuction.sol

```diff
@@ -21,17 +23,21 @@ contract BatchAuction is Ownable {
         bool applied;
     }
     
-    uint public depositIndex;
     mapping (uint => DepositState) public depositHashes;
 
     event Deposit(uint16 accountId, uint8 tokenIndex, uint amount, uint slot);
-    event IncrementDepositIndex(uint index);
+    event StateTransistion(string transitionType, bytes32 from, bytes32 to);
 
     modifier onlyRegistered() {
         require(publicKeyToAccountMap[msg.sender] != 0, "Must have registered account");
         _;
     }
 
+    constructor () public {
+        // The initial state should be Pederson hash of an empty balance tree
+        stateRoots.push(0);
```

Comment:
Do you know how to create this state?

Reply:
I was gonna wait till Monday when Alex provides me with it. Although eventually I will need to code this up for my JS-scripts to continue with unit testing. Also, I don't know how to do this.

Reply:
I created https://github.com/gnosis/dex-zksnarks/pull/40 which should make it easy.

Do you really need a working implementation in JS? I think as far as the unit tests are concerned, since there is no actual verification inside the smart contract, you can give it any values.

Reply:
Yea, but once we get to testing challenges, it might help to have some small examples.